src/Symfony/Component/Console/Color.php

```diff
@@ -202,4 +205,62 @@ private function rgbToHex(string $color): string
 
         return '#'.implode('', $hexString);
     }
+
+    /**
+     * Explained formula can be found here: {@see https://en.wikipedia.org/wiki/HSL_and_HSV#HSL_to_RGB}.
+     */
+    private function hslToHex(string $color): string
+    {
+        if (!preg_match(self::HSL_FUNCTIONAL_NOTATION_REGEX, $color, $matches)) {
+            throw new InvalidArgumentException(sprintf('Invalid HSL functional notation; should be of the form "hsl(h, s%%, l%%)", got "%s".', $color));
+        }
+
+        [$hue, $saturation, $lightness] = \array_slice($matches, 1);
+
+        if ($hue > 359) {
+            throw new InvalidArgumentException(sprintf('Invalid hue; value should be between 0 and 359, got %d.', $hue));
```

Comment:
Why this exception ? In CSS, `hsl(4236, 50%, 50%)` is valid. `4236deg` is equivalent to `276deg` as hue as far as the color is concerned (but it may make a difference regarding transitions, I'm not sure)